zkevm-circuits/src/evm_circuit/execution/logs.rs

```diff
@@ -126,8 +126,7 @@ impl<F: Field> ExecutionGadget<F> for LogGadget<F> {
         );
 
         let copy_rwc_inc = cb.query_cell();
-        let dst_addr = (1u64 << 32).expr() * TxLogFieldTag::Data.expr()
-            + (1u64 << 48).expr() * (cb.curr.state.log_id.expr() + 1.expr());
+        let dst_addr = build_tx_log_expression(0, TxLogFieldTag::Data, cb.curr.state.log_id);
```

Comment:
When used with the function that only takes expressions, this would look like:

`build_tx_log_expression(1.expr(), TxLogFieldTag::Data.expr(), cb.curr.state.log_id)`

Note that originally this `dst_addr` used `1` as index.  I don't know if that's correct, but I'd leave it the same to keep the same behavior.

Reply:
Actually I got mistaken, that `1` is not the index, it's actually that the log_id value is `cb.curr.state.log_id.log_id + 1`!  Sorry about that!